build_tools/hack/build_env_diag.py

```diff
@@ -18,1123 +19,2202 @@
+            tester.check_MAX_PATH,
+            tester.check_py,
+            tester.check_git,
+            tester.check_gitlfs,
+            tester.check_uv,
+            tester.check_cmake,
+            tester.check_ccache,
+            tester.check_ninja,
+            tester.check_gfortran,
+            tester.check_VS20XX,
+            tester.check_msvc,
+            tester.check_VC_HOST,
+            tester.check_VC_TARGET,
+            tester.check_ml64,
+            tester.check_lib,
+            tester.check_link,
+            tester.check_VC_SDK,
+            tester.check_rc,
+        ]
+
+    elif device.LINUX:
+        test = [
+            tester.check_Device_OS,
+            tester.check_Device_ARCH,
+            tester.check_DISK_USAGE,
+            tester.check_py,
+            tester.check_git,
+            tester.check_gitlfs,
+            tester.check_uv,
+            tester.check_cmake,
+            tester.check_ccache,
+            tester.check_ninja,
+            tester.check_gcc,
+            tester.check_gxx,
+            tester.check_gfortran,
+            tester.check_gcc_as,
+            tester.check_gcc_ar,
+            tester.check_ld,
+        ]
+
+    for items in test:
+        _stat, _except, _result = items.__call__()
+        print(f"{_stat} {_except}")
+        tester.check_record.append(_result)
+
+    check_summary(tester)
+
+
+# Define Main() as main diagnosis, Help() as `--help`, Track() as `--track`.
+def main():
+
+    therock_detect_start = time.perf_counter()
+
+    # os.system("cls" if os.name == "nt" else "clear")  Disabled clear prompt.
+
+    TheRock.__logo__()
+
+    device = Device()
+    device.summary
 
-    traceback: Disk space tool small or disk usage too high
-    """,
-            Color_Warning,
-        )
+    print(
+        r"""
+        ===========    Start detecting conpoments for building ROCm\TheRock    ===========
+    """
+    )
+
+    run_test()
 
-therock_detect_terminate = time.perf_counter()
-therock_detect_time = therock_detect_terminate - therock_detect_start
-print(
-    f"""
-    ===========    TheRock build pre-diagnosis script completed in {hint(f"{therock_detect_time:.2f}")} seconds    ===========
-"""
-)
+    therock_detect_terminate = time.perf_counter()
+    therock_detect_time = float(therock_detect_terminate - therock_detect_start)
+    therock_detect_runtime = cstring(f"{therock_detect_time:2f}", "hint")
+    print(
+        f"""
+        ===========    TheRock build pre-diagnosis script completed in {therock_detect_runtime} seconds    ===========
+    """
+    )
+
+    time.sleep(0.5)
+
+
+# Possiable Args.
+help_args = ("--help", "-help", "-?", "/help", "/?", "--i-need-help")
+
+
+# Launcher.
+if __name__ == "__main__":
+    if len(sys.argv) == 1:
+        main()
+    else:
+        args = [arg.lower() for arg in sys.argv[1:]]
+        if any(arg in help_args for arg in args):
+            TheRock.help()
+        else:
+            TheRock.help()
```

Comment:
Per my other comment about help text, I would drop this code and change to just

```python
if __name __ == "__main__":
  main()
```

We can use argparse with some help text handler if we want, without anything custom for things like `/help`